File: hw/palomo/u3/hw22MazeDiagramClass/MazeApplication/src/main/java/ec/edu/espe/mazeapplication/controller/MazeController.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package ec.edu.espe.mazeapplication.controller;

import ec.edu.espe.mazeapplication.model.Maze;
/**
 *
 * @author Cristian Palomo, Error 404, @ESPE
 */
public class MazeController {
    private Maze maze;

    public void createMaze(int rows, int columns) {

        maze = new Maze(rows, columns);

        MazeGenerator generator = new MazeGenerator(maze);

        generator.createRooms();
        generator.assignWalls();
        generator.assignDoors();

    }

    public Maze getMaze() {
        return maze;
    }

    public String drawMaze() {

        String text = "";

        for (int i = 0; i < maze.getRooms().length; i++) {

            for (int j = 0; j < maze.getRooms()[0].length; j++) {

                text += "[ ]";

            }

            text += "\n";

        }

        return text;

    }
}

File: hw/palomo/u3/hw22MazeDiagramClass/MazeApplicationPython/controller/maze_controller.py
from model.maze import Maze
from controller.random_maze_generator import RandomMazeGenerator


class MazeController:

    def __init__(self):
        self._maze = None

    def create_maze(self, n, m):
        self._maze = Maze(n, m)
        generator = RandomMazeGenerator(self._maze)
        generator.generate_maze()

    def initialize_maze(self):
        pass

    def get_maze(self):
        return self._maze

    def draw_maze(self):
        drawing = ""

        for row in self._maze.rooms:
            for room in row:
                drawing += "[ ]"
            drawing += "\n"

        return drawing